e2e/etcd_release_upgrade_test.go

```diff
@@ -86,3 +91,55 @@ func TestReleaseUpgrade(t *testing.T) {
 		}
 	}
 }
+
+func TestReleaseUpgradeWithSnapshot(t *testing.T) {
```

Comment:
```
Set snap count to 10.
Write 50 keys.
upgrade cluster.
kill the upgraded cluster without doing any more v3 writes
restart the upgraded cluster and verify everything is OK
```

1. we should write 50 keys instead of 10 to ensure a snapshot is triggered
2. after the final restart, we need to check the cluster works